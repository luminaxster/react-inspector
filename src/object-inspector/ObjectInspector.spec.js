import TestRenderer from 'react-test-renderer';
import {createRoot} from 'react-dom/client';
import {act} from 'react-dom/test-utils';
import ObjectInspector from './ObjectInspector';

describe('ObjectInspector', () => {
   it('should render', () => {
      const tree = TestRenderer.create(<ObjectInspector/>);
      expect(tree).toMatchSnapshot();
   });
   
   it('passes `nodeRenderer` prop to <TreeView/>', () => {
      const nodeRenderer = () => <span>unit test</span>;
      
      const tree = TestRenderer.create(
         <ObjectInspector nodeRenderer={nodeRenderer}/>
      );
      
      expect(tree).toMatchSnapshot();
   });
});

describe('ObjectInspector Content', () => {
   let container;
   let prev;
   
   beforeAll(()=>{
      prev = global.IS_REACT_ACT_ENVIRONMENT;
      global.IS_REACT_ACT_ENVIRONMENT = true;
   });
   
   afterAll(()=>{
      global.IS_REACT_ACT_ENVIRONMENT = prev;
   });
   
   beforeEach(() => {
      container = document.createElement('div');
      document.body.appendChild(container);
   });
   
   afterEach(() => {
      document.body.removeChild(container);
      container = null;
   });
   
   it('should render with Maps with Regex and Maps keys', () => {
      const data = new Map([
         [/\S/g, 'Regular Expression key']
      ]);
      
      
      act(() => {
         createRoot(container).render(<ObjectInspector data={data}/>);
      });
      
      const button = container.querySelector('div');
      
      act(() => {
         button.dispatchEvent(new MouseEvent('click', {bubbles: true}));
      });
      
      expect(container.innerHTML).toMatchSnapshot();
   });
})
