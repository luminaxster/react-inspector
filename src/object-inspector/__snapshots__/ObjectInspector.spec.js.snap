// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectInspector Content should render with Maps with Regex and Maps keys 1`] = `"<ol role=\\"tree\\" class=\\"css-1mbgsgl\\"><li aria-expanded=\\"true\\" role=\\"treeitem\\" class=\\"css-9vhyes\\"><div class=\\"css-0\\"><span class=\\"css-18ss45a-Arrow\\">â–¶</span><span class=\\"css-gkg3jw-ObjectPreview\\">Map </span><span class=\\"css-gkg3jw-ObjectPreview\\">{}</span></div><ol role=\\"group\\" class=\\"css-18j2x0a\\"><li aria-expanded=\\"false\\" role=\\"treeitem\\" class=\\"css-9vhyes\\"><div class=\\"css-0\\"><span class=\\"css-26z203\\">&nbsp;</span><span><span class=\\"css-1n87jha-ObjectValue\\">/\\\\S/g</span><span>: </span><span class=\\"css-1n87jha-ObjectValue\\">\\"Regular Expression key\\"</span></span></div><ol role=\\"group\\" class=\\"css-18j2x0a\\"></ol></li></ol></li></ol>"`;

exports[`ObjectInspector passes \`nodeRenderer\` prop to <TreeView/> 1`] = `
<ol
  className="css-1mbgsgl"
  role="tree"
>
  <li
    aria-expanded={false}
    className="css-9vhyes"
    role="treeitem"
  >
    <div
      className="css-0"
      onClick={[Function]}
    >
      <span>
        unit test
      </span>
    </div>
    <ol
      className="css-18j2x0a"
      role="group"
    />
  </li>
</ol>
`;

exports[`ObjectInspector should render 1`] = `
<ol
  className="css-1mbgsgl"
  role="tree"
>
  <li
    aria-expanded={false}
    className="css-9vhyes"
    role="treeitem"
  >
    <div
      className="css-0"
      onClick={[Function]}
    >
      <span
        className="css-utwc52-ObjectValue"
      >
        undefined
      </span>
    </div>
    <ol
      className="css-18j2x0a"
      role="group"
    />
  </li>
</ol>
`;
